src/app/ui/components/containers/modal/components/radix-dialog.tsx

```diff
@@ -0,0 +1,54 @@
+import { ReactNode } from 'react';
+
+import * as Dialog from '@radix-ui/react-dialog';
```

Comment:
I replaced `BaseDrawer` with [Radix Dialog](https://www.radix-ui.com/primitives/docs/components/dialog) as our drawers are actually modal / dialogs anyway. 

I am using the `primitive` here but we use the `theme` version in `AllowDiagnosticsLayout`. 

I have more work to do investigating if we can use the same one for both. 

Overall it gives us an upgrade on coding it ourself as:
- it is more responsive out of the box
- it handles `ESC` by default
- it has built in accessibility features

Reply:
Curious about your import/naming here bc I'm working on the dropdown, are you thinking we should always use `Radix` in the component name when customizing a radix component to share?

Reply:
I did this as I copied it out of the Radix docs this way. I can change it to destructure it, it's not a big deal:
`import { Content, Overlay, Portal, Root } from '@radix-ui/react-dialog';`

I was thinking its good to use `Radix` in the name if you agree? 

Here I have a wrapper for `@radix-ui/react-dialog` that I called `RadixDialog` that customises it to meet our needs. 

I then include it from another component called `Modal` which on consideration I think I should rename to `Dialog`. 

That could be a good approach, what do you think?

So for your work you'd have `RadixDropdown` being consumed by `DropDown`. 

We may not always need to do that. In this case, there is some common code needed for all Modals/ Dialogs so I needed to wrap the radix component (for `navigate`). I could probably refactor further to just have one component called `Dialog`

Reply:
I don't have a strong opinion here. I just wanted to follow your lead if it was a pattern you were starting.

Reply:
I suppose it is best to consume components in our code, from the `ui` folder, that avoid using `Radix` in the name?

Reply:
Sounds great to me and it's good to keep in sync 👍 

I may be able to use the `theme` dialog so I will test that in case we can avoid adding the primitive